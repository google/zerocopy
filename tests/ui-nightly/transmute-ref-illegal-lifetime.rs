// Copyright 2023 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

fn main() {}

fn increase_lifetime() {
    let x = 0u64;
    // It is illegal to increase the lifetime scope.
    let _: &'static u64 = zerocopy::transmute_ref!(&x);
}
