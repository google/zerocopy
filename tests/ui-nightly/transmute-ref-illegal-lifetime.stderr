error[E0597]: `x` does not live long enough
  --> tests/ui-nightly/transmute-ref-illegal-lifetime.rs:10:52
   |
8  |     let x = 0u64;
   |         - binding `x` declared here
9  |     // It is illegal to increase the lifetime scope.
10 |     let _: &'static u64 = zerocopy::transmute_ref!(&x);
   |            ------------                            ^^ borrowed value does not live long enough
   |            |
   |            type annotation requires that `x` is borrowed for `'static`
11 | }
   | - `x` dropped here while still borrowed
